packages/core/src/lib/social.ts

```
@@ -40,17 +39,17 @@ export const getUserInfoByConnectorCode = async (
 export const getUserInfoFromInteractionResult = async (
   connectorId: string,
   interactionResult?: InteractionResults
-): Promise<SocialUserInfo> => {
+): Promise<SocialUserInfo | null> => {
   const result = z
     .object({
       socialUserInfo: z.object({
         connectorId: z.string(),
         userInfo: socialUserInfoGuard,
       }),
     })
-    .parse(interactionResult);
-
-  assertThat(result.socialUserInfo.connectorId === connectorId, 'session.connector_id_mismatch');
+    .safeParse(interactionResult);
 
-  return result.socialUserInfo.userInfo;
+  return result.success && result.data.socialUserInfo.connectorId === connectorId
```

show throw error instead of returning null